File: hw/rojas/u2/hw19Inheritance/hw19InheritancePythonAndJavaScript/hw20Python/ec.edu.espe.model/Cow.py
from FarmAnimal import FarmAnimal

class Cow(FarmAnimal):
    def __init__(self, id, breed, born_on, gender, is_able_to_reproduce, weight, cage, is_producing_milk, litters_a_day):
        super().__init__(id, breed, born_on, gender, is_able_to_reproduce, weight, cage)
        self.is_producing_milk = is_producing_milk
        self.litters_a_day = litters_a_day

    def milk(self):
        if self.is_producing_milk:
            print(f"Milking... Got {self.litters_a_day} liters.")
            return self.litters_a_day
        else:
            print("This cow is not producing milk right now.")
            return 0

    def __str__(self):
        return (f"--> COW:\n"
                f"    ID: {self.id}, Breed: {self.breed}, Weight: {self.weight}kg\n"
                f"    Born On: {self.born_on.date()}, Gender: {self.gender}\n"
                f"    Can Reproduce: {self.is_able_to_reproduce}, Location: {self.cage}\n"
                f"    [Specific] Producing Milk: {self.is_producing_milk}, Liters/Day: {self.litters_a_day}")

File: hw/chalan/u2/hw19Inheritance/FarmsystemJS/model/Cow.js
const FarmAnimal = require('./FarmAnimal');

class Cow extends FarmAnimal {
    constructor(isProducingMilk, litersPerDay, id, breed, bornOn, gender, isAbleToReproduce, weight, cage) {
        super(id, breed, bornOn, gender, isAbleToReproduce, weight, cage);
        this.isProducingMilk = isProducingMilk;
        this.litersPerDay = litersPerDay;
    }
    toString() { return `Cow{liters=${this.litersPerDay}, ${super.toString()}}`; }
}
module.exports = Cow;